client/received_blocks_new_map.go

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package client
+
+import (
+	"github.com/cespare/xxhash"
+	"github.com/m3db/m3x/ident"
+	"github.com/m3db/m3x/pool"
+)
+
+func newReceivedBlocksMap(pool pool.BytesPool) *receivedBlocksMap {
+	return _receivedBlocksMapAlloc(_receivedBlocksMapOptions{
+		hash: func(k idAndBlockStart) receivedBlocksMapHash {
+			return receivedBlocksMapHash(xxhash.Sum64(k.id.Bytes())) +
```

Comment:
hm might be better for distribution to offset the addition, i.e. 

```
hash: func(k idAndBlockStart) receivedBlocksMapHash {
  const SOME_PRIME_NUMBER = 7
  return receivedBlocksMapHash(k.blockStart * SOME_PRIME_NUMBER ) +
    receivedBlocksMapHash(xxhash.Sum64(k.id.Bytes()))
 }

Reply:
Fair, I'll use the standard Java composite key approach.